### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by thomas-w-grant** [webpro/knip#97]
I am running `npx knip` with no config on a nx workspace.

I am getting the following error:

```
Analyzing source files...
file:///Users/tomgrant/.nvm/versions/node/v16.17.0/lib/node_modules/knip/dist/dependency-deputy.js:108
                if (referencedDependencies?.has(dependency))
                                            ^

RangeError: Maximum call stack size exceeded
    at Set.has (<anonymous>)
    at isUnreferencedDependency (file:///Users/tomgrant/.nvm/versions/node/v16.17.0/lib/node_modules/knip/dist/dependency-deputy.js:108:45)
    at file:///Users/tomgrant/.nvm/versions/node/v16.17.0/lib/node_modules/knip/dist/dependency-deputy.js:123:70
    at Array.find (<anonymous>)
    at isUnreferencedDependency (file:///Users/tomgrant/.nvm/versions/node/v16.17.0/lib/node_modules/knip/dist/dependency-deputy.js:123:46)
    at file:///Users/tomgrant/.nvm/versions/node/v16.17.0/lib/node_modules/knip/dist/dependency-deputy.js:123:70
    at Array.find (<anonymous>)
    at isUnreferencedDependency (file:///Users/tomgrant/.nvm/versions/node/v16.17.0/lib/node_modules/knip/dist/dependency-deputy.js:123:46)
    at file:///Users/tomgrant/.nvm/versions/node/v16.17.0/lib/node_modules/knip/dist/dependency-deputy.js:123:70
    at Array.find (<anonymous>)
```

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.10.0. See [Release 2.10.0](https://github.com/webpro/knip/releases/tag/2.10.0) for release notes._

**Comment by webpro** [comment]
I think I found the culprit: circular peer dependencies indeed. Hopefully no more infinite recursion :)
